### PR TITLE
Wip/animations

### PR DESCRIPTION
**Pull request by MattVsTheWorld**
Added support for:

- Character outline before objects (required extra materials)
- Rotation of character model
- Deathzone below the world (deleted walls from scene to test it). Death restart game scene
- Animations for idle and walk
- Shooting projectiles in front of character

### CODE REVIEW

File: assets/3d/zahir/materials/zahir-brown-base.tres
Comment (dughiboogie) [inline assets/3d/zahir/materials/zahir-brown-base.tres:1]: Missing CamelCase

File: assets/3d/zahir/materials/zahir-brown.tres
Comment (dughiboogie) [inline assets/3d/zahir/materials/zahir-brown.tres:1]: Missing CamelCase

File: assets/3d/zahir/materials/zahir-teal-base.tres
Comment (dughiboogie) [inline assets/3d/zahir/materials/zahir-teal-base.tres:1]: Missing CamelCase

File: assets/3d/zahir/materials/zahir-teal.tres
Comment (dughiboogie) [inline assets/3d/zahir/materials/zahir-teal.tres:1]: Missing CamelCase

File: assets/3d/zahir/materials/zahir-white-base.tres
Comment (dughiboogie) [inline assets/3d/zahir/materials/zahir-white-base.tres:1]: Missing CamelCase

File: assets/3d/zahir/materials/zahir-white.tres
Comment (dughiboogie) [inline assets/3d/zahir/materials/zahir-white.tres:1]: Missing CamelCase

File: project.godot [hunk 1]
```diff
@@ -15,6 +15,11 @@ run/main_scene="res://scenes/PrototypeScene_01.tscn"
 config/features=PackedStringArray("4.2", "Forward Plus")
 config/icon="res://icon.svg"
 
+[display]
+
+window/size/viewport_width=1440
+window/size/viewport_height=900
+
```
Comment (dughiboogie) [inline project.godot]:
```suggestion
```

File: scripts/Player.gd [hunk 2]
```diff
@@ -16,11 +15,31 @@ const JUMP_VELOCITY = 5.0
...
+
+# ------------------------------------
 # Get the gravity from the project settings to be synced with RigidBody nodes.
 var gravity = ProjectSettings.get_setting("physics/3d/default_gravity")
 
+#func _ready():
+	## Pre-initialize orientation transform.
+	#orientation = player_model.global_transform
+	#orientation.origin = Vector3()
+
```
Comment (dughiboogie) [inline scripts/Player.gd]:
```suggestion

```

File: scripts/Player.gd [hunk 3]
```diff
@@ -55,8 +73,55 @@ func _physics_process(delta):
...
+		_animate(ANIMATIONS.IDLE, delta)
+	else:
+		_animate(ANIMATIONS.WALK, delta)
+		
+	# --- Shooting ---
+	if Input.is_action_just_pressed("shoot"):
+		var shoot_origin = shoot_from.global_transform.origin
+		var shoot_dir = shoot_to.global_transform.origin
+		# Spawn and shoot bullet
+		var bullet = preload("res://scenes/Projectile.tscn").instantiate()
```
Comment (dughiboogie) [inline scripts/Player.gd]: Preload Projectile scene in file header

File: scripts/Player.gd [hunk 4]
```diff
@@ -55,8 +73,55 @@ func _physics_process(delta):
...
+
+		var current_blend: float = animation_tree["parameters/Idle2Walk/blend_amount"]
+		if not current_blend == 1:
+			var new_blend: float = min(current_blend + delta*MOTION_INTERPOLATE_SPEED, 1)
+			animation_tree["parameters/Idle2Walk/blend_amount"] = new_blend
+
+func _shoot():
+	var shoot_particle = $PlayerModel/Armature/Skeleton3D/BulletOrigin/ShootParticle
+	shoot_particle.restart()
+	shoot_particle.emitting = true
```
Comment (dughiboogie) [inline scripts/Player.gd]: Find out what this does

File: scripts/Projectile.gd [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+var hit = false
+
+@onready var collision_shape = $CollisionShape3D
+
+
+
+
+func _physics_process(delta):
+	if hit:
+		# TODO: re-enable once it seems ok
```
Comment (dughiboogie) [inline scripts/Projectile.gd]:
```suggestion

```

**Review by dughiboogie — CHANGES_REQUESTED** [review]
Some changes needed, cool pew pew

**Comment by MattVsTheWorld** [comment]
Should have fixed all issues

**Review by dughiboogie — APPROVED** [review]
A-very a-good!

**Pull request merged**